Mounting secrets in pods (#703)
lightbend/cloudflow: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration.adoc

It also shows how to set options to the JVM through an environment variable
and how to add labels to a pod. 

> It also possible to add `volumes`, that contain secrets, to the pods. And then mount this volumes into the container with `volume-mounts`. Bear in mind that each `volume-mounts` name has to match a `volumes`  name, defined in the same config file. Here's a sample of how to set two `volumes` with two different secrets and how to mount them in a container.  

Reviewer: ```suggestion
It is also possible to add `volumes`, that contain secrets, to the pods. And then mount these volumes into the container with `volume-mounts`. Bear in mind that each `volume-mounts` name has to match a `volumes`  name, defined in the same config file. Here's a sample of how to configure two `volumes` with two different secrets and how to mount them in the container.  The streamlet code can access the secret data in `mysecret` in the file `/mnt/folderA/mysecret`
```

Reviewer: Not sure if that is exactly right but would be nice to provide this detail here

[source, hocon]
----
cloudflow.streamlets.my-streamlet {
  kubernetes.pods.pod {
   volumes {
      foo {
        secret {
          name = mysecret
        }
      }
      bar {
        secret {
          name = mysecret2
        }
      }
    }
    containers.container {
      volume-mounts {
        foo {
          mountPath = "/mnt/folderA"
          readOnly = true
        }
        bar {
          mountPath = "/mnt/folderB"
          readOnly = true
        }
      }
    }
  } 
}
----

> This configuration will mount `mysecret` into each `my-streamlet` container on to the path `/mnt/folderA` as read only, while `mysecret2` will be mounted on path `/mnt/folderB`. Each `volume-mounts` name is matching a `volumes` name as `volume-mounts.foo` matches `volumes.foo` and `volume-mounts.bar` matches `volumes.bar`

Reviewer: ```suggestion
This configuration will mount `mysecret` into each `my-streamlet` container on to the path `/mnt/folderA` as read only, while `mysecret2` will be mounted on path `/mnt/folderB`. Each `volume-mounts` name is matching a `volumes` name as `volume-mounts.foo` matches `volumes.foo` and `volume-mounts.bar` matches `volumes.bar`.
```

Author: is `Each `volume-mounts` name must matching a `volumes` name` right @RayRoestenburg shouldn't it be `Each `volume-mounts` name must match a `volumes` name`

It also shows how to set options to the JVM through an environment variable
and how to add labels to a pod. 

> It also possible to add `volumes` to pods where the source of the volume is a secret. Then these can be mounted into the pod with `volume-mounts`. 

Reviewer: ```suggestion
It also possible to add `volumes` to pods where the source of the volume is a secret. Then these can be mounted into the container with `volume-mounts`. 
```

> There is an exception for `volume-mounts` regards being abel to set and specific type of pod. Will cover that later on, now here below, a sample of how to configure two `volumes` with secrets and how to mount them.  

Reviewer: This first sentence needs some work.

Author: done

[source, hocon]
----
cloudflow.streamlets.my-streamlet {
  kubernetes.pods.pod {
   volumes {
      foo {
        secret {
          name = mysecret
        }
      }
      bar {
        secret {
          name = mysecret2
        }
      }
    }
    containers.container {
      volume-mounts {

>         foo {

Reviewer: I think we need to be a bit more specific that a name under volume-mounts (the name of a volume-mount) must match a name under volumes  (the name of the volume).

Author: done

          mountPath = "/mnt/folderA"
          readOnly = true
        }
        bar {
          mountPath = "/mnt/folderB"
          readOnly = true
        }
      }
    }
  } 
}
----

> This configuration will mount `mysecret` into each `my-streamlet` pod on path `/mnt/folderA` as read only. While `mysecret2` will be mounted into `/mnt/folderB` . Bare in mind that each `volume-mount` has to point to a `volume` defined in the same config.

Reviewer: ```suggestion
This configuration will mount `mysecret` into each `my-streamlet` pod on path `/mnt/folderA` as read only, while `mysecret2` will be mounted on path `/mnt/folderB` . Bear in mind that each `volume-mount` has to point to a `volume` defined in the same config.
```

The above example shows specific container settings for the Spark driver and executor pods. In this case the driver pods will request `1` cpu and `1Gi` of memory, limited to `2Gi` of memory and will be labeled with `key1 : value1` and 
`key2 : value2`. The executors will be requesting `2` cpus, `2Gi` of memory, limited to `6Gi` and will be labeled with `key3 : value3`.

> Regards `volumes` and `volume-mounts`, `runtime.spark` is the only runtime that allows to set different `volume-mounts` to each of its pods. More concretly, while the `volumes` have to be set under `cloudflow.runtime.spark.kubernetes.pods.pod` the 'volumen-mounts' can be set specifically on the `cloudflow.runtime.spark.kubernetes.pods.driver` and/or the `cloudflow.runtime.spark.kubernetes.pods.executor`. The sample below shows how to:

Reviewer: ```suggestion
Regarding `volumes` and `volume-mounts`, `runtime.spark` is the only runtime that allows to set different `volume-mounts` to each of its pods. More concretely, while the `volumes` have to be set under `cloudflow.runtime.spark.kubernetes.pods.pod` the 'volume-mounts' can be set specifically on the `cloudflow.runtime.spark.kubernetes.pods.driver` and/or the `cloudflow.runtime.spark.kubernetes.pods.executor`. The sample below shows how to:
```

[source, hocon]
----
cloudflow.runtime.spark {
  kubernetes.pods {
    pod {
     volumes {
        foo {
          secret {
            name = mysecret
          }
        }
        bar {
          secret {
            name = mysecret2
          }
        }
      }
    }
    driver {
      containers.container {
        volume-mounts {
          foo {
            mountPath = "/mnt/file1"
            readOnly = true
          }
        }
      }
    } 
    executor {
      containers.container {
        volume-mounts {
          bar {
            mountPath = "/mnt/file2"
            readOnly = true
          }
        }
      }
    } 
  }
}
---- 

> This configuration will mount `mysecret` into any `driver` pod on path `/mnt/file1` as read only. While `mysecret2` will be mounted into any `executor` pod on path `/mnt/file2`.

Reviewer: This is likely confusing since file1 will be the path under which more files exist, right? rather change the names of the paths.

Author: is confusing

The above example shows specific container settings for the Spark driver and executor pods. In this case the driver pods will request `1` cpu and `1Gi` of memory, limited to `2Gi` of memory and will be labeled with `key1 : value1` and 
`key2 : value2`. The executors will be requesting `2` cpus, `2Gi` of memory, limited to `6Gi` and will be labeled with `key3 : value3`.

> Regards `volumes` and `volume-mounts` Spark runtime is the only runtime that allows to set different `volume-mounts` to each of its pods. Which are `driver` and `executor`. This will work as the following. The `volumes` are set in every pod but it's the `volume-mounts` that can be made specific to a type of pod. The sample below show how to:

Reviewer: I think we should rewrite this text a little..

Author: done
